gcovr/coverage.py

```diff
@@ -140,7 +148,7 @@ def update(self, other):
         assert self.lineno == other.lineno
         self.count += other.count
         self.noncode &= other.noncode
-
+        self.excluded &= other.excluded
```

Comment:
Is it correct to exclude a line only if all coverage data sets being merged exclude this line? Or should a line be excluded if any data set excludes it?

Conflicting exclusions could appear e.g. if we're merging two JSON files, one of which was created with `--no-markers` or if they had different `--exclude-lines-by-pattern` settings.

I think that if one data set has the `excluded` flag set, then this means that data on that line should not be considered. I would write this as `self.excluded |= other.excluded`, but am open to different arguments. For example, using an `&` would be consistent with how file-level filtering works.

This is different from the `noncode` flag which we use to mark lines that gcov considers to be uncovered – if another coverage data set does include data for that line, the noncode flag should disappear.

If this line is changed, then the code for merging file coverage below must not set `excluded=True`.

Reply:
A good question. I think also that excluding the line one time will exclude it in the merged report is better. Else we would get lower coverage values. I'll change this.